src/ir/auth_logic/auth_logic_ast_traversing_visitor_test.cc

```diff
@@ -86,102 +87,102 @@ class TraversalOrderVisitor
     return result;
   }
 
   Unit PreVisit(const Fact& fact) override {
-    if (pre_visits_) nodes_.push_back(fact.DebugPrint());
+    if (pre_visits_) nodes_.push_back(std::addressof(fact));
     return Unit();
   }
   Unit PostVisit(const Fact& fact, Unit result) override {
-    if (post_visits_) nodes_.push_back(fact.DebugPrint());
+    if (post_visits_) nodes_.push_back(std::addressof(fact));
     return result;
   }
 
   Unit PreVisit(const ConditionalAssertion& condAssertion) override {
-    if (pre_visits_) nodes_.push_back(condAssertion.DebugPrint());
+    if (pre_visits_) nodes_.push_back(std::addressof(condAssertion));
     return Unit();
   }
   Unit PostVisit(const ConditionalAssertion& condAssertion,
                  Unit result) override {
-    if (post_visits_) nodes_.push_back(condAssertion.DebugPrint());
+    if (post_visits_) nodes_.push_back(std::addressof(condAssertion));
     return result;
   }
 
   Unit PreVisit(const Assertion& assertion) override {
-    if (pre_visits_) nodes_.push_back(assertion.DebugPrint());
+    if (pre_visits_) nodes_.push_back(std::addressof(assertion));
     return Unit();
   }
   Unit PostVisit(const Assertion& assertion, Unit result) override {
-    if (post_visits_) nodes_.push_back(assertion.DebugPrint());
+    if (post_visits_) nodes_.push_back(std::addressof(assertion));
     return result;
   }
 
   Unit PreVisit(const SaysAssertion& saysAssertion) override {
-    if (pre_visits_) nodes_.push_back(saysAssertion.DebugPrint());
+    if (pre_visits_) nodes_.push_back(std::addressof(saysAssertion));
     return Unit();
   }
   Unit PostVisit(const SaysAssertion& saysAssertion, Unit result) override {
-    if (post_visits_) nodes_.push_back(saysAssertion.DebugPrint());
+    if (post_visits_) nodes_.push_back(std::addressof(saysAssertion));
     return result;
   }
 
   Unit PreVisit(const Query& query) override {
-    if (pre_visits_) nodes_.push_back(query.DebugPrint());
+    if (pre_visits_) nodes_.push_back(std::addressof(query));
     return Unit();
   }
   Unit PostVisit(const Query& query, Unit result) override {
-    if (post_visits_) nodes_.push_back(query.DebugPrint());
+    if (post_visits_) nodes_.push_back(std::addressof(query));
     return result;
   }
 
   Unit PreVisit(const Program& program) override {
-    if (pre_visits_) nodes_.push_back(program.DebugPrint());
+    if (pre_visits_) nodes_.push_back(std::addressof(program));
     return Unit();
   }
   Unit PostVisit(const Program& program, Unit result) override {
-    if (post_visits_) nodes_.push_back(program.DebugPrint());
+    if (post_visits_) nodes_.push_back(std::addressof(program));
     return result;
   }
 
-  const std::vector<std::string>& nodes() const { return nodes_; }
+  const std::vector<const void*>& nodes() const { return nodes_; }
 
  private:
   bool pre_visits_;
   bool post_visits_;
-  std::vector<std::string> nodes_;
+  std::vector<const void*> nodes_;
```

Comment:
`void*` should basically never be used. This should be either some base class of the possibilities or, if you don't care about dereferencing the pointer and only want it as a number to be compared/hased/etc, an `intptr_t` or `uintptr_t`.

Reply:
resolved